xsuite/src/world/csproxyBin.ts

```diff
@@ -0,0 +1,63 @@
+import { ChildProcess } from "node:child_process";
+import {
+  getCsproxyBinPath,
+  getCsproxyDefaultConfigPath,
+} from "@xsuite/chainsimulator";
+import { spawnChildProcess } from "./childProcesses";
+
+export const startCsproxyBin = async (
+  debug: boolean = false,
+  configFilePath?: string,
+  nodeOverrideFilePath?: string,
+): Promise<{
+  server: ChildProcess;
+  proxyUrl: string;
+}> => {
+  const csproxyBinPath = getCsproxyBinPath();
+  const csproxyConfigFolder = getCsproxyDefaultConfigPath();
+
+  const server = spawnChildProcess(`${csproxyBinPath}`, [
+    "--server-port",
+    "0",
+    "--config",
+    configFilePath || `${csproxyConfigFolder}/config.toml`,
+    "--node-override-config",
+    nodeOverrideFilePath || `${csproxyConfigFolder}/nodeOverride.toml`,
+    "--node-configs",
+    `${csproxyConfigFolder}/node/config`,
+    "--proxy-configs",
+    `${csproxyConfigFolder}/proxy/config`,
+  ]);
+
+  server.stderr.on("data", (data: Buffer) => {
+    throw new Error(data.toString());
+  });
+
+  server.on("error", (error) => {
+    throw error;
+  });
+
+  const proxyUrl = await new Promise<string>((resolve, reject) => {
+    server.stdout.on("data", (data: Buffer) => {
+      if (debug) {
+        console.log(data.toString());
+      }
+
+      const timeout = setTimeout(
```

Comment:
Now that the MvX team has improved the chain simulator, do we still need this?

Reply:
Yes because the Chain Simulator displays multiple logs, then hangs for 1-2 seconds, then the URL is displayed.
Without the timeout the `match` would just fail every time.

Reply:
Why would it fail? It will not resolve while the URL is not displayed. Still don't see why this is needed

Reply:
This should be removed

Reply:
Done